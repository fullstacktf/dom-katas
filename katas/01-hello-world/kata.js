export function kata() {
  const div = document.createElement("div");
  div.id = "square";
  const p = document.createElement("p");
  p.id = "text";
  p.textContent = "Hello World";

  div.appendChild(p);

  document.body.appendChild(div);
}
