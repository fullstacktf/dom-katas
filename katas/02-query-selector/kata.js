export function kata() {
  const starWars = document.querySelector("main > section:nth-child(2) > p");
  starWars.classList.add("star-wars");
}
