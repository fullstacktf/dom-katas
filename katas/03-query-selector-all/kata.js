export function kata() {
  const titles = document.querySelectorAll("section > h2");

  for (const title of titles) {
    title.classList.add("title");
  }
}
