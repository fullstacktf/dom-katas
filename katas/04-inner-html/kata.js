const page = `<main>
<article>
    <img src="https://picsum.photos/id/28/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>Trip Planning for Total Beginners</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
<article>
    <img src="https://picsum.photos/id/35/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>7 ways to survive in the desert</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
<article>
    <img src="https://picsum.photos/id/37/800/300">
    <div class="content">
        <p class="category">travel</p>
        <h2>5 amazing travel destinations</h2>
        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum amet reprehenderit nesciunt
            repellendus
            dolorem ipsa eaque repudiandae odit aperiam est assumenda, impedit alias doloremque temporibus
            eligendi
            neque, qui quas esse.</p>
    </div>
</article>
</main>`;

export function kata() {
  document.body.innerHTML = page;
}
