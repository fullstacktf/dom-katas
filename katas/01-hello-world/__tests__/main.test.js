// query utilities:
import {
  getByLabelText,
  getByText,
  getByTestId,
  queryByTestId,
  // Tip: all queries are also exposed on an object
  // called "queries" which you could import here as well
  waitFor,
} from "@testing-library/dom";

import { kata } from "../kata.js";

function getExampleDOM() {
  // This is just a raw example of setting up some DOM
  // that we can interact with. Swap this with your UI
  // framework of choice ðŸ˜‰
  const div = document.createElement("div");
  div.innerHTML = `
      <label for="username">Username</label>
      <input id="username" />
      <button>Print Username</button>
    `;
  const button = div.querySelector("button");
  const input = div.querySelector("input");
  button.addEventListener("click", () => {
    // let's pretend this is making a server request, so it's async
    // (you'd want to mock this imaginary request in your unit tests)...
    setTimeout(() => {
      const printedUsernameContainer = document.createElement("div");
      printedUsernameContainer.innerHTML = `
          <div data-testid="printed-username">${input.value}</div>
        `;
      div.appendChild(printedUsernameContainer);
    }, Math.floor(Math.random() * 200));
  });
  return div;
}

test("has hello world string", async () => {
  const root = kata();

  getByText(root, "Hello World");
});

test("structure is ok", async () => {
  const root = kata();

  expect(root).toMatchSnapshot();
});
