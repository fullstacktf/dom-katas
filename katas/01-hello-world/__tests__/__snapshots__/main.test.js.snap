// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`structure is ok 1`] = `
<body>
  
    

  <div
    id="square"
  >
    <p
      id="text"
    >
      Hello World
    </p>
  </div>
</body>
`;
